File: CsharpProjects/StateSolucao03/StateSolucao03/domains/state/AguardandoPagamentoState.cs
using StateSolucao03.domains;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StateSolucao03.domains.state
{
    internal class AguardandoPagamentoState : State
    {
        private Pedido pedido;
        public AguardandoPagamentoState(Pedido pedido)
        {
            this.pedido = pedido;
        }
        void State.cancelarPedido()
        {
            this.pedido.estadoAtual = new CanceladoState(pedido);
        }
        void State.despacharPedido()
        {
            throw new Exception("Operacao não suportada, " +
                "pedido ainda não foi pago");
        }
        void State.sucessoAoPagar()
        {
            this.pedido.estadoAtual = new PagoSate(pedido);
        }
    }
}

File: JavaProjects/StateSolucao03/src/domains/state/AguardandoPagamentoState.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package domains.state;

import domains.Pedido;
import java.util.logging.Level;
import java.util.logging.Logger;

/**
 *
 * @author jeffe
 */
public class AguardandoPagamentoState implements State{
    
    private Pedido pedido;

    public AguardandoPagamentoState(Pedido pedido) {
        this.pedido = pedido;
    }

    @Override
    public void sucessoAoPagar() {
        this.pedido.setEstadoAtual(new PagoState(pedido));
    }

    @Override
    public void cancelarPedido() {
        this.pedido.setEstadoAtual(new CanceladoState(pedido));
    }

    @Override
    public void despacharPedido() {
        try {
            throw new Exception("Operação não suportada - pedido ainda não foi pago");
        } catch (Exception ex) {
            Logger.getLogger(AguardandoPagamentoState.class.getName()).log(Level.SEVERE, null, ex);
        }
    }
}
